substrate/frame/treasury/src/lib.rs

```diff
@@ -1020,6 +1029,72 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			// Must never be less than 0 but better be safe.
 			.saturating_sub(T::Currency::minimum_balance())
 	}
+
+	/// Ensure the correctness of the state of this pallet.
+	///
+	/// ## Invariants
+	///
+	/// 1. [`ProposalCount`] >= [`Proposals`].count(). The [`ProposalCount`] SV is only increased, but never
+	///    decreased whereas individual proposals can be removed from the [`Proposals`] SM.
+	/// 2. [`T::ProposalBondMinimum`] * Number of proposals with non-zero bond <= sum_{p in Proposals} p.bond.
+	/// 3. (Optional) [`T::ProposalBondMaximum`] * Number of proposals with non-zero bond >= sum_{p in Proposals} p.bond.
+	/// 4. Each [`ProposalIndex`] contained in [`Approvals`] should exist in [`Proposals`]. Note, that this
+	///    automatically implies Approvals.count() <= Proposals.count().
+	/// 5. [`SpendCount`] >= [`Spends`].count(). Follows the same reasoning as (1).
+	/// 6. For each spend entry contained in `Spends` we should have spend.expire_at > spend.valid_from.
+	#[cfg(any(feature = "try-runtime", test))]
+	fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+		ensure!(
+			ProposalCount::<T, I>::get() as usize >= Proposals::<T, I>::iter().count(),
```

Comment:
Same thing that i wrote about the `SpendCount` applies here; akk `Proposal` indices must be smaller than it.